istio/istio.go

```diff
@@ -113,6 +115,43 @@ func (istio *Istio) ApplyOperation(ctx context.Context, opReq adapter.OperationR
 			ee.Details = ""
 			hh.StreamInfo(e)
 		}(istio, e)
+	case internalconfig.PrometheusAddon, internalconfig.GrafanaAddon, internalconfig.KialiAddon, internalconfig.JaegerAddon, internalconfig.ZipkinAddon:
+		go func(hh *Istio, ee *adapter.Event) {
+			_, err := hh.InstallAddon(opReq.Namespace, opReq.IsDeleteOperation, opReq.OperationName)
+			operation := "install"
+			if opReq.IsDeleteOperation {
+				operation = "uninstall"
+			}
+
+			if err != nil {
+				e.Summary = fmt.Sprintf("Error while %sing %s", operation, opReq.OperationName)
+				e.Details = err.Error()
+				hh.StreamErr(e, err)
+				return
+			}
+			ee.Summary = fmt.Sprintf("Succesfully %sed %s", operation, opReq.OperationName)
+			ee.Details = fmt.Sprintf("Succesfully %sed %s from the %s namespace", operation, opReq.OperationName, opReq.Namespace)
+			hh.StreamInfo(e)
```

Comment:
If we're able to pass along the URL to the new endpoint created for the user to access the add-on... that'd be helpful. 

Not as helpful, though, as adding Kiali and Jaeger to the list of things MeshSync v1 discovers.

Reply:
@leecalcote, we have a function in meshkit called `GetServiceEndpoint` which returns the address for the service. We can achieve the same thing in the MeshSync v1 by tweaking it a bit. But, both kiali and jaeger addon manifest only gives us a cluster IP based endpoint by default. 
Would that be fine? 

Also, we have a [generic expose function](https://github.com/layer5io/meshery-linkerd/pull/160) in linkerd adapter which can expose these services and give them a `nodeport`/`loadbalancer` based address, so we can copy that function here or maybe move it to meshkit, if we want to expose these services too.

Another appoach could be to modify the default addon manifest and change service type to `NodePort`/`LoadBalancer`, I am not sure if tweaking the original manifest would be a good solution or not :sweat_smile: .

Any suggestions?

Reply:
Fantastic list. Let's get the `expose` function bubble up to meshkit and then back down across each of the adapters. 👍